# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/dietlibc/pkg_patch/pkg_gcc3.patch
# ROCK Linux is Copyright (C) 1998 - 2004 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./libiberty/regex.c~	Fri Nov 16 22:54:17 2001
+++ ./libiberty/regex.c	Sun Dec 29 05:39:31 2002
@@ -302,7 +302,7 @@
 
 /* Integer type for pointers.  */
 # if !defined _LIBC && !defined HAVE_UINTPTR_T
-typedef unsigned long int uintptr_t;
+/*typedef unsigned long int uintptr_t;*/
 # endif
 
 /* Should we use malloc or alloca?  If REGEX_MALLOC is not defined, we
--- oldgcc/gcc/fixinc/server.c	Wed Apr  4 00:46:26 2001
+++ newgcc/gcc/fixinc/server.c	Mon Jan  6 15:43:34 2003
@@ -204,7 +204,8 @@
 
   fputs ("trap : 1\n", server_pair.pf_write);
   fflush (server_pair.pf_write);
-  p_cur_dir = getcwd ((char *) NULL, MAXPATHLEN + 1);
+  p_cur_dir = (char *) xmalloc(MAXPATHLEN + 1);
+  getcwd (p_cur_dir, MAXPATHLEN + 1);
 }
 
 /*
